pyscriptjs/src/components/pyrepl.ts

```diff
@@ -51,6 +51,50 @@ export function make_PyRepl(interpreter: InterpreterClient, app: PyScriptApp) {
             logger.debug(`element ${this.id} successfully connected`);
         }
 
+        get src() {
+            return this.getAttribute('src');
+        }
+
+        set src(value) {
+            this.setAttribute('src', value);
+        }
+
+        attributeChangedCallback(name: string, oldVal: string, newVal: string) {
+            if (name === 'src' && newVal !== oldVal) {
+                void this.loadReplSrc();
+            }
+        }
+
+        /**
+         * Fetch url from src attribute of py-repl tags and
+         * preload the code from fetch response into the Corresponding py-repl tag,
+         * but please note that they will not be pre-run unless you click the runbotton.
+         */
+        async loadReplSrc() {
+            await fetch(this.src)
+                .then(response => {
+                    if (response.ok) {
+                        return response.text();
+                    } else {
+                        logger.error(
+                            `loading code from ${this.src} to repl...Failed ${response.status} (${response.statusText})`,
+                        );
+                    }
+                })
+                .then(content => {
+                    if (content !== void 0) {
+                        const code = content.toString().replace(/[<&]/g, match => (match === '<' ? '&lt;' : '&amp;'));
```

Comment:
in no circumstance `response.text()` returns a non string so `toString()` there looks redundant: https://developer.mozilla.org/en-US/docs/Web/API/Response/text#return_value